### PR TITLE
Feature/blue green partitioning

### PR DESCRIPTION
**Pull request by 138bit**
Blue/Green partition

### CODE REVIEW

File: arch/arm/mach-bcm283x/Kconfig [hunk 1]
```diff
@@ -196,8 +196,16 @@ config TARGET_RPI_ARM64
 	  the RPi 4 model B, in AArch64 (64-bit) mode.
 	select ARM64
 
+config TARGET_FANCOM_HMI
+	bool "Fancom HMI based on CM4"
+	help
+	  fancom
```
Comment (UberDoefus) [inline arch/arm/mach-bcm283x/Kconfig:202]: Maybe place a more descriptive explanation on the help part?

File: board/fancom/hmi/Kconfig [hunk 2]
```diff
@@ -0,0 +1,22 @@
+if TARGET_FANCOM_HMI
+
+config SYS_BOARD
+        default "hmi"
+
+config SYS_VENDOR
+        default "fancom"
```
Comment (UberDoefus) [inline board/fancom/hmi/Kconfig:7]: Maybe Fancom iso fancom

File: board/fancom/hmi/MAINTAINERS [hunk 3]
```diff
@@ -0,0 +1,6 @@
+RPI BOARD
```
Comment (UberDoefus) [inline board/fancom/hmi/MAINTAINERS:1]: Copy paste? we're the maintaners right?

File: board/fancom/hmi/hmi.c [hunk 4]
```diff
@@ -0,0 +1,514 @@
...
+}
+
+int ft_board_setup(void *blob, struct bd_info *bd)
+{
+//	/*
+//	 * For now, we simply always add the simplefb DT node. Later, we
+//	 * should be more intelligent, and e.g. only do this if no enabled DT
+//	 * node exists for the "real" graphics driver.
+//	 */
+//	lcd_dt_simplefb_add_node(blob);
```
Comment (UberDoefus) [inline board/fancom/hmi/hmi.c:505]: Why has this been commented out?, then the CONFIG_RPI_EFI_NR_SPIN_PAGES doesn't make sense anymore in the Kconfig

File: board/fancom/hmi/hmi.c [hunk 5]
```diff
@@ -0,0 +1,514 @@
...
+
+	snprintf(serial_string, sizeof(serial_string), "%016llx",
+		 msg->get_board_serial.body.resp.serial);
+	env_set("serial#", serial_string);
+}
+
+int misc_init_r(void)
+{
+	set_fdt_addr();
+	//set_fdtfile();
```
Comment (UberDoefus) [inline board/fancom/hmi/hmi.c:412]: Why has this been commented out?

File: board/fancom/hmi/hmi.c [hunk 6]
```diff
@@ -0,0 +1,514 @@
...
+	env_set("fdtfile", fdtfile);
+}
+
+/*
+ * If the firmware provided a valid FDT at boot time, let's expose it in
+ * ${fdt_addr} so it may be passed unmodified to the kernel.
+ */
+static void set_fdt_addr(void)
+{
+	//if (env_get("fdt_addr"))
```
Comment (UberDoefus) [inline board/fancom/hmi/hmi.c:321]: Why has this been commented out?

File: include/configs/hmi.h [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+#include <linux/sizes.h>
+#include <asm/arch/timer.h>
+#include <linux/stringify.h>
+
+#ifndef __ASSEMBLY__
+#include <asm/arch/base.h>
+#endif // __ASSEMBLY__
+
+#define CONFIG_SYS_TIMER_RATE		1000000
```
Comment (UberDoefus) [inline include/configs/hmi.h:17]: Apparently, the base was not the rpi.h file https://github.com/fancom/u-boot/blob/fancom/master/include/configs/rpi.h, but why not? That file differs from what we have here...

**Review by JvanDooren — CHANGES_REQUESTED** [review]
Please address my remarks and questions

File: board/fancom/hmi/MAINTAINERS (hunk 3 above)
Comment (138bit) [reply board/fancom/hmi/MAINTAINERS:1]: Yes this was from copy/paste. I disregarded it but will fix this.

File: board/fancom/hmi/Kconfig (hunk 2 above)
Comment (138bit) [reply board/fancom/hmi/Kconfig:7]: Nope, because than have to rename the directory in boards/.

File: arch/arm/mach-bcm283x/Kconfig (hunk 1 above)
Comment (138bit) [reply arch/arm/mach-bcm283x/Kconfig:202]: done

File: include/configs/hmi.h (hunk 7 above)
Comment (138bit) [reply include/configs/hmi.h]: Fixed.

File: board/fancom/hmi/hmi.c (hunk 4 above)
Comment (138bit) [reply board/fancom/hmi/hmi.c:505]: This was part of a test to get blue/green working. I will reenable and test it.

File: board/fancom/hmi/hmi.c (hunk 5 above)
Comment (138bit) [reply board/fancom/hmi/hmi.c:412]: This was part of a test to get blue/green working. I will reenable and test it.

File: board/fancom/hmi/hmi.c (hunk 6 above)
Comment (138bit) [reply board/fancom/hmi/hmi.c]: Because is messes up our FDT.

**Comment by 138bit** [comment]
> Please address my remarks and questions

Done

**Review by JvanDooren — APPROVED** [review]
Looks ok

**Pull request merged**